### PR TITLE
feat(documentation): add form-select stories

### PR DESCRIPTION
**Pull request by oliverschuerch**

### CODE REVIEW

**Comment by swisspost-devs** [comment]
Preview environments are getting posted here, as soon as they are ready!

**Review by gfellerph — CHANGES_REQUESTED** [review]

File: packages/documentation/.storybook/preview-head.html [hunk 1]
```diff
@@ -6,9 +6,17 @@
...
     font-size: 100%;
   }
+
+  #docs-root code {
+    margin: 0;
+    padding: 0;
+    background: none;
+    border: 0 none;
+    color: #e03c31;
+  }
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/.storybook/preview-head.html:19]: To fullfill the AA standards a rating from 4.5 would be enough and we're very close to this here.

I would prefer something like I implemented here. It's how bootstrap is doing it, and their docs are well styled/formatted and pretty nice to read.

If we want a more github like styling of inline code elements, I'd prefer a light color an a dark background;
image
image

What do you think?

File: packages/documentation/src/stories/components/select/select.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+- You can define the attribute `selected` directly on the `option` element.
+
+<Canvas>
+  <Story id="components-select--selected-option" />
+</Canvas>
+
+### Disabled Options
+
+It is possible to disable individual `option` elements. Disabled options cannot be selected by the user.<br/>
+However, it is possible to deactivate the initially selected option as well. This option can then not be reselected after the user changed the selection once.
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.docs.mdx:59]: This is only the halfe truth...
As a user of a screen reader which is focusing a select field I would probably not press "space" or "enter" to open the dropdown menu of the select field but only select one of them with the arrow keys. In this case, the screenreader does not say "option 1 of 5" but only reads the option label (e.g. "option 1").

And with all this accessability topcis in mind, I think we should not forget, that about 75-95% of or users are not blind or visually impaired. All the seeing people should be able to visually recognize, that the option is disabled.

And in my opinion, disabled options in a select or a checkbox/radio button group are definitly a common use cases.

[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/documentation/src/stories/components/select/select.docs.mdx:59]: What would be a reason to disable options and how are you communicating this to the user?

File: packages/documentation/.storybook/preview-head.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/documentation/.storybook/preview-head.html:19]: It's personal preference, but I'm not a fan of the bootstrap style and I like the light variant better as it's a more subtle.

File: packages/documentation/src/stories/components/select/select.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+  args: {
+    label: 'Label',
+    options: [
+      { text: 'Elektu opcion...' },
+      { text: 'Opcion 1', value: 1 },
+      { text: 'Opcion 2', value: 2 },
+      { text: 'Opcion 3', value: 3 },
+      { text: 'Opcion 4', value: 4 },
+      { text: 'Opcion 5', value: 5 },
+    ] as ISelectOption[],
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.stories.tsx:36]: This is only the halfe truth...
As a "blind" person which is focusing a select field I would probably not press "space" or "enter" to open the dropdown menu of the select field but only select one of them with the arrow keys. In this case, the screenreader does not say "option 1 of 5" but only reads the option label (e.g. "option 1").

And with all this accessability topcis in mind, I think we should not forget, that about 75-95% of or users are not blind or visually impaired.

And in my opinion, disabled options in a select, a checkbox or a radio button group are definitly a common use cases.

File: packages/documentation/src/stories/components/select/select.docs.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.docs.mdx:59]: It's not about reasons, it's about giving developers all the options.
Of course, we can remove this example, but the developers out there know about the `disabled` attribute and after all, we implement appropriate styles for it.
So what would be the reason to hide this information?

As long as our styles package is not up to date to properly work with the `readonly` attribute (which could be the accessable alternative to the `disabled` attribute), I'll still prefer to show this example.

And yes, we can tell the developers to not disable options dynamically, but just omit them completely.
But how would you communicate that runtime change to disabled users?
In my opinion, the problem stays the same, but one of the cases is a better approach for non disabled users.

[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.docs.mdx:59]: We decided to just integrate a hint, that the disabled state on form-fields has some accessability issues and a link to the describing part in the accessability docs page.

[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.docs.mdx]: There was one sentence from you @gfellerph that make me look with an other angle on the docs:
"We do only document our Design System, not how HTML or some elements work".

Thats why I removed the option-stories completly. As every developer should be able to find out, how to set the value of a `select` element, without the need of our documentation to show this.

File: packages/documentation/src/stories/components/select/select.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.stories.tsx]: Totally removed this in the newest version ;)

File: packages/documentation/src/stories/components/select/select.docs.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+The `select` can be used by simply adding one of four classes:
+
+- Small: `.form-select-sm`
+- Regular: `.form-select-rg`
+- Medium: default or `.form-select-md`
+- Large: `.form-select-lg`
+
+<div class="alert alert-info alert-md">
+  Do not use this classes on a <code>select</code> elements wrapped within a{' '}
```
Comment (gfellerph) [inline packages/documentation/src/stories/components/select/select.docs.mdx:64]:
```suggestion
  Do not use this classes on <code>select</code> elements wrapped within a{' '}
```

File: packages/documentation/src/stories/components/select/select.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    hiddenLabel: {
+      name: 'Hidden Label',
+      description: '<p>Render the component with or without a visible label.</p><div className="alert alert-info alert-sm">There are accessability issues with hidden labels.<br/>Please read our <a href="/?path=/story/foundations-accessability--page#labels">labels accessability guide</a>.</div>',
```
Comment (gfellerph) [inline packages/documentation/src/stories/components/select/select.stories.tsx:56]:
```suggestion
      description: '<p>Render the component with or without a visible label.</p><div className="alert alert-info alert-sm">There are accessibility issues with hidden labels.<br/>Please read our <a href="/?path=/story/foundations-accessibility--page#labels">labels accessibility guide</a>.</div>',
```

File: packages/documentation/src/stories/components/select/select.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    disabled: {
+      name: 'Disabled',
+      description: '<p>When set to `true`, disables the component\'s functionality and places it in a disabled state.</p><div className="alert alert-info alert-sm">There are accessability issues with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessability--page#disabled-state">disabled state accessability guide</a>.</div>',
```
Comment (gfellerph) [inline packages/documentation/src/stories/components/select/select.stories.tsx:144]:
```suggestion
      description: '<p>When set to `true`, disables the component\'s functionality and places it in a disabled state.</p><div className="alert alert-info alert-sm">There are accessibility issues with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</div>',
```

File: packages/documentation/src/stories/foundation/accessability.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { Meta } from '@storybook/addon-docs';
+
+<Meta
+  title="Foundations/Accessability"
```
Comment (gfellerph) [inline packages/documentation/src/stories/foundation/accessability.stories.mdx:4]:
```suggestion
  title="Foundations/Accessibility"
```

File: packages/documentation/src/stories/foundation/accessability.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  parameters={{
+    previewTabs: {
+      canvas: {
+        hidden: true,
+      },
+    },
+  }}
+/>
+
+# Accessability
```
Comment (gfellerph) [inline packages/documentation/src/stories/foundation/accessability.stories.mdx:14]:
```suggestion
# Accessibility
```

File: packages/documentation/src/stories/foundation/accessability.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+<div
+  style={{
+    margin: '0 0 1rem',
+    padding: '1rem',
+    background: 'red',
+    fontWeight: 'bold',
+    color: 'white',
+  }}
+>
+  TODO: write accessability page
```
Comment (gfellerph) [inline packages/documentation/src/stories/foundation/accessability.stories.mdx:25]:
```suggestion
  TODO: write accessibility page
```

File: packages/documentation/src/stories/foundation/accessability.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+</div>
+
+## Forms
+
+### Labels
+
+TODO: write accessability guide about form labels (visible and invisible), aria-label, ...
```
Comment (gfellerph) [inline packages/documentation/src/stories/foundation/accessability.stories.mdx:32]:
```suggestion
TODO: write accessibility guide about form labels (visible and invisible), aria-label, ...
```

File: packages/documentation/src/stories/foundation/accessability.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+### Disabled state
+
+TODO: write accessability guide about disabled states on input, textarea, select, select-options,
```
Comment (gfellerph) [inline packages/documentation/src/stories/foundation/accessability.stories.mdx:36]:
```suggestion
TODO: write accessibility guide about disabled states on input, textarea, select, select-options,
```

**Review by gfellerph — APPROVED** [review]
Just typos and text changes, otherwise nice story!

**Pull request merged**